third_party/move/move-binary-format/src/builders.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    access::ModuleAccess,
+    errors::{PartialVMError, PartialVMResult},
+    file_format::*,
+};
+use move_core_types::{
+    account_address::AccountAddress, identifier::IdentStr, language_storage::ModuleId,
+    vm_status::StatusCode,
+};
+
+fn get_or_add<T: PartialEq>(pool: &mut Vec<T>, val: T) -> PartialVMResult<usize> {
```

Comment:
A bit sad to see the caching gone here. Without caching it could be `O(n^2)` complexity to build a complex script. If another component on the production path picks this up it could also run into time complexity issues unexpectedly, albeit not at the same severity as the resource viewer.